workflow/ignite/decorators/evaluate.py

```diff
@@ -1,21 +1,32 @@
-from torch import no_grad
+import torch
 from functools import wraps
 
+from workflow.functional import structure_map
 from workflow.torch import model_device
 from workflow.ignite.decorators.to_device import to_device
 
 
+def detach(x):
+    if type(x) is torch.Tensor:
```

Comment:
if something is a subtype it should also detach
https://docs.python.org/3/library/functions.html#issubclass

Reply:
Yes maybe, I think we should use `isinstance` though. I can change that

Reply:
sorry, mixed it up, still a bit rusty in the weirder parts of python :P
yes isinstance is the most pythonic way I think

also I think there where a few more instances of is torch.Tensor :)

Reply:
The reason that I didn't use isinstance is that I'm not sure that we actually want to allow subclasses everywhere but I might be wrong

For example in `structure_map` we do not want to allow subclasses

Reply:
subclasses of Tensor should still be a Tensor in all senses of the type, otherwise they have implemented it badly

as for inside `structure_map` the issue is that you want to construct the same type as you deconstruct
so yes, with this implementation you would want to keep it as == equals to maintain the property that the structure is preserved

it would be possible to use the dynamic nature of python and have
```
if isinstance(d, dict):
    return type(d)(map(lambda kv: (kv[0], function(kv[1])), d.items()))
```

Reply:
this would support `OrderedDict` for example and other things that are sub of dict (as long as they have .items() and are able to construct from [(k, v)]

Reply:
Yes I considered what you're saying but I didn't want to make it too complicated. Maybe we should reevaluate that but I think it's good enough in the current condition and we can make a PR to change that if we want to

Reply:
yes I think so to, as stated "in the current implementation"
I wasn't requesting a change for what structure_map supports
it will be backwards compatible if we add support for subtypes of the types listed later :)